src/SportsHub.Infrastructure/Repositories/TeamRepository.cs

```diff
@@ -79,5 +79,35 @@ public async Task<bool> DoesTeamAlredyExistByIdAsync(Guid id)
 
             return teams.Any(team => team.Id == id);
         }
+
+        public IEnumerable<Team> GetTeamsFilteredByLocation(string location, ICollection<Team> teams)
+        {
+            return teams.Where(teams => teams.Location == location).ToList();
+        }
+
+        public IEnumerable<Team> GetTeamsFilteredBySubcategoryId(Guid subcategoryId, ICollection<Team> teams)
+        {
+            return teams.Where(teams => teams.SubcategoryId == subcategoryId).ToList();
+        }
+
+        public IEnumerable<Team> GetTeamsFilteredBySubcategoryIds(IQueryable<Subcategory> subcategories, ICollection<Team> teams)
+        {
+            var subcategoryIds = (from subcategory in subcategories
+                                  select subcategory.Id).ToList();
+            var newTeam = teams;
+            teams = newTeam.Where(team => team.SubcategoryId == subcategoryIds[0]).ToList();
+
+            for (int i = 1; i < subcategoryIds.Count; i++)
+            {
+                teams = teams.Concat(newTeam.Where(team => team.SubcategoryId == subcategoryIds[i]).ToList()).ToList();
+            }
+
+            return teams.AsEnumerable();
+        }
+
+        public async Task<IEnumerable<Team>> GetSortedTeamAsync()
+        {
+            return await _context.Teams.OrderBy(teams => teams.Name).ToListAsync();
```

Comment:
Why do you need a separate method for ordering?

Reply:
I made a separate sort method referring to code from FilterArticles

Reply:
But you can just add ordering when returning teams by get request